bin/deploy

```diff
@@ -261,53 +268,57 @@ main() {
-    # installation attempt.  These need to be cleaned up before [re]attempting
-    # to install packages.
-    conda clean -y --all
-    if [[ $deploy_mode == "bleed" ]]; then
-      ARGS=()
-      ARGS+=('env' 'update')
-      ARGS+=('--name' "$LSST_CONDA_ENV_NAME")
-      ARGS+=("--file" "$env_file")
-      conda "${ARGS[@]}"
-    else
-      ARGS=()
-      ARGS+=('create')
-      ARGS+=('--name' "$LSST_CONDA_ENV_NAME")
-      ARGS+=('-y')
-      ARGS+=("--file" "$lock_file")
-      # check if env exists and is consistent with the lock file
-      tmp_lock="$(mktemp)"
-      if (conda run -n "${LSST_CONDA_ENV_NAME}" conda list --explicit > "${tmp_lock}"); then
-        # check if environment is unchanged
-        # -B ignores blank lines
-        if (diff -B "${tmp_lock}" "${lock_file}"); then
-          echo "Environment exists and is consistent with its definition"
+  if conda env list | grep "^$LSST_CONDA_ENV_NAME " > /dev/null && [[ $CHECKENV = false ]]; then
+    echo "::: Environment $LSST_CONDA_ENV_NAME already deployed."
+  else
+    (
+      echo "::: Deploy new environment $LSST_CONDA_ENV_NAME"
+      # Install packages on which the stack is known to depend
+      # conda may leave behind lock files from an uncompleted package
+      # installation attempt.  These need to be cleaned up before [re]attempting
+      # to install packages.
+      conda clean -y --all
+      if [[ $deploy_mode == "bleed" ]]; then
+        ARGS=()
+        ARGS+=('env' 'update')
+        ARGS+=('--name' "$LSST_CONDA_ENV_NAME")
+        ARGS+=("--file" "$env_file")
+        conda "${ARGS[@]}"
+      else
+        ARGS=()
+        ARGS+=('create')
+        ARGS+=('--name' "$LSST_CONDA_ENV_NAME")
+        ARGS+=('-y')
+        ARGS+=("--file" "$lock_file")
+        # check if env exists and is consistent with the lock file
+        tmp_lock="$(mktemp)"
+        if (conda run -n "${LSST_CONDA_ENV_NAME}" conda list --explicit > "${tmp_lock}"); then
+          # check if environment is unchanged
+          # -B ignores blank lines
+          if (diff -B "${tmp_lock}" "${lock_file}"); then
+            echo "Environment exists and is consistent with its definition"
+          else
+            # environment has changed, it will be recreated
+            conda "${ARGS[@]}"
+          fi
         else
-          # environment has changed, it will be recreated
+          # environment do not exist, it will be created
           conda "${ARGS[@]}"
         fi
-      else
-        # environment do not exist, it will be created
-        conda "${ARGS[@]}"
+        rm "${tmp_lock}"
       fi
-      rm "${tmp_lock}"
-    fi
 
-    # disable the conda install progress bar when not attached to a tty. Eg.,
-    # when running under CI
-    if [[ ! -t 1 ]]; then
-      ARGS+=("--quiet")
-    fi
+      # disable the conda install progress bar when not attached to a tty. Eg.,
+      # when running under CI
+      if [[ ! -t 1 ]]; then
+        ARGS+=("--quiet")
+      fi
```

Comment:
Is this tabs vs spaces (and correct?)

Reply:
I don't think there should be tabs here. I just indented 2 spaces more, to keep (visual) consistency with the rest, but I assume it makes no difference.